tests/unittest_inference.py

```diff
@@ -3897,6 +3897,72 @@ class Clazz(metaclass=_Meta):
         ).inferred()[0]
         assert isinstance(cls, nodes.ClassDef) and cls.name == "Clazz"
 
+    def test_infer_subclass_attr_outer_class(self):
+        node = extract_node(
+            """
+        class Outer:
+            data = 123
+
+        class Test(Outer):
+            pass
+        Test.data
+        """
+        )
+        inferred = next(node.infer())
+        self.assertIsInstance(inferred, nodes.Const)
+        self.assertEqual(inferred.value, 123)
+
+    def test_infer_subclass_attr_inner_class_works_from_classdef(self):
+        node = extract_node(
+            """
+        class Outer:
+            class Inner:
+                data = 123
+
+        class Test(Outer.Inner):
+            pass
+        Test
+        """
+        )
+        cls_def = next(node.infer())
+        self.assertIsInstance(cls_def, nodes.ClassDef)
+        inferred = next(cls_def.igetattr("data"))
+        self.assertIsInstance(inferred, nodes.Const)
+        self.assertEqual(inferred.value, 123)
+
+    def test_infer_subclass_attr_inner_class_works_indirectly(self):
+        node = extract_node(
+            """
+        class Outer:
+            class Inner:
+                data = 123
+        Inner = Outer.Inner
+
+        class Test(Inner):
+            pass
+        Test.data
+        """
+        )
+        inferred = next(node.infer())
+        self.assertIsInstance(inferred, nodes.Const)
+        self.assertEqual(inferred.value, 123)
+
+    def test_infer_subclass_attr_inner_class(self):
```

Comment:
This test is very similar to the second one you added, where you test the `Test` class. I wonder if it would be a good idea to make only one test. Thus if one of the assertion fails and not the other it could be easier to understand the origin of the bug. Instead with two separated tests, if one fails we won't notice there is another one very similar which is successful. 
Another idea is to just add a comment indicating that another test is similar.
What do you think about it?

Reply:
My usual style has been to have tests as "small" as possible, each one showing just the minimum change it needs for differing behaviour. In this case, accessing the element via `inferred = next(cls_def.igetattr("data"))` worked before the fix was applied, but accessing the element via `next(extract_node(..."Test.data").infer()` did not, which to me proved that this was a bug. I can certainly combine `test_infer_subclass_attr_inner_class_works_from_classdef` and `test_infer_subclass_attr_inner_class` with a comment highlighting the different behaviour with and without the fix so that it's clear that this is a non-regression test.